proctor-common/src/main/java/com/indeed/proctor/common/ProctorUtils.java

```diff
@@ -1417,4 +1440,29 @@ private static boolean evaluatePayloadValidator(
 
         return false;
     }
+
+    private static boolean isUnitlessAllocation(@Nonnull final Allocation allocation) {
+        final int unitlessIdentifierIndex =
+                allocation.getRule().indexOf(UNITLESS_ALLOCATION_IDENTIFIER);
```

Comment:
nit: i know we already check allocation.getRule() != null in the calling code. However it may not be a bad idea to check again here.

Reply:
Removed, but had to remove Nonnull annotation as IDE/spotbugs will complain.